tests/integration/piv_test.py

```diff
@@ -1391,22 +1391,212 @@ def test_piv_added_stop_time_by_event(mock_rabbitmq):
+    res = api_post("/piv/{}".format(PIV_CONTRIBUTOR_ID), data=ujson.dumps(piv_23187_addition))
+    assert "PIV feed processed" in res.get("message")
+
+    _assert_db_added_stop_time_in_the_middle()
+
+
+def test_piv_re_routed_at_the_beginning(mock_rabbitmq):
+    piv_23187_addition = _get_stomp_20201022_23187_stop_time_added_at_the_beginning_fixture()
+    piv_23187_addition["objects"][0]["object"]["evenement"] = [
+        {"type": "MODIFICATION_DETOURNEMENT", "texte": None}
+    ]
+    disruption = DisruptionTuple(type=None, texte=None)
+    modification = ModificationTuple(statut="CREATION_DETOURNEMENT", motif=None)
     _set_event_on_stop(
         fixture=piv_23187_addition,
         dep_or_arr_key="depart",
-        rang=3,
+        rang=0,
         disruption=disruption,
         modification=modification,
     )
-    _set_event_on_stop(
+    _set_event_on_stops(
         fixture=piv_23187_addition,
-        dep_or_arr_key="arrivee",
-        rang=3,
+        rang_min=1,
+        rang_max=1,
+        disruption=disruption,
+        modification=modification,
+    )
+    modification = ModificationTuple(statut=None, motif=None)
+    disruption = DisruptionTuple(type="SUPPRESSION_DETOURNEMENT", texte=None)
+    _set_event_on_stops(
+        fixture=piv_23187_addition,
+        rang_min=2,
+        rang_max=2,
         disruption=disruption,
         modification=modification,
     )
 
     res = api_post("/piv/{}".format(PIV_CONTRIBUTOR_ID), data=ujson.dumps(piv_23187_addition))
     assert "PIV feed processed" in res.get("message")
 
-    _assert_db_added_stop_time_in_the_middle()
+    with app.app_context():
+        assert RealTimeUpdate.query.count() == 1
+        assert TripUpdate.query.count() == 1
+        db_trip = TripUpdate.query.first()
+
+        assert db_trip.status == "update"
+        assert db_trip.effect == "DETOUR"
+        assert len(db_trip.stop_time_updates) == 8
+
+        added_st = db_trip.stop_time_updates[0]
+        assert added_st.stop_id == "stop_point:PIV:85010116:Train"
+        assert added_st.departure_status == ModificationType.add.name
+        assert added_st.departure == datetime(2020, 10, 22, 20, 10)
+
+        added_st = db_trip.stop_time_updates[1]
+        assert added_st.stop_id == "stop_point:PIV:85010082:Train"
+        assert added_st.arrival_status == ModificationType.add.name
+        assert added_st.arrival == datetime(2020, 10, 22, 20, 22)
+        assert added_st.departure_status == ModificationType.add.name
+        assert added_st.departure == datetime(2020, 10, 22, 20, 24)
+
+        deleted_st = db_trip.stop_time_updates[2]
+        assert deleted_st.stop_id == "stop_point:PIV:85010231:Train"
+        assert deleted_st.arrival_status == ModificationType.delete.name
+        assert deleted_st.departure_status == ModificationType.delete.name
+
+        for s in db_trip.stop_time_updates[3:]:
+            assert s.arrival_status == "none"
+            assert s.departure_status == "none"
```

Comment:
Should we use this instead?
```suggestion
            assert s.arrival_status == ModificationType.none.name
            assert s.departure_status == ModificationType.none.name
```